inference-engine/src/vpu/common/src/ngraph/operations/static_shape_non_maximum_suppression.cpp

```diff
@@ -64,15 +83,34 @@ StaticShapeNonMaxSuppression::clone_with_new_inputs(const OutputVector& new_args
 }
 
 void StaticShapeNonMaxSuppression::validate_and_infer_types() {
-    ngraph::op::v4::NonMaxSuppression::validate_and_infer_types();
+    ngraph::op::v5::NonMaxSuppression::validate_and_infer_types();
+
+    const auto boxesPS = get_input_partial_shape(0);
+    const auto scoresPS = get_input_partial_shape(1);
+
+    auto outShape = get_output_partial_shape(0);
+
+    if (boxesPS.rank().is_static() && scoresPS.rank().is_static()) {
+        const auto numBoxesDim = boxesPS[1];
+        const auto maxOutputBoxesPerClassNode = input_value(2).get_node_shared_ptr();
+        if (numBoxesDim.is_static() && scoresPS[0].is_static() && scoresPS[1].is_static() &&
+            ngraph::op::is_constant(maxOutputBoxesPerClassNode)) {
+            const auto numBoxes = numBoxesDim.get_length();
+            const auto numClasses = scoresPS[1].get_length();
+            const auto maxOutputBoxesPerClass = max_boxes_output_from_input();
+
+            outShape[0] = std::min(numBoxes, maxOutputBoxesPerClass) * numClasses *
+                                    scoresPS[0].get_length();
+        }
+    }
```

Comment:
* If `ngraph::op::v5::NonMaxSuppression::validate_and_infer_types` provides us with static output shape - should we do nothing? Should replacement NMS v5 -> StaticShapeNMS be avoided in such cases?
* Code here is unclear:
  * There are a lot of checks if something is static and then do the job. Should we propagate upper-bound only in case of dynamism? So checks should be in form of *if something is dynamic then do the job*? If current check fails - we do nothing and hopefully fail with ambiguous message *something is wrong* below. If your intention is *if I have to propagate upper-bound and some dimension/shape does not provide me information for that* - be explicit and say it - which dimension/shape
  * What `scoresPS[0]` and `scoresPS[1]` mean? Do you try to check if `scoresPS` in common is static?
  * I guess you upper-bound estimation is based on specification. I expect `ngraph::op::v5::NonMaxSuppression::validate_and_infer_types` does something similar or even the same in case of static form. Is it possible to reuse code, maybe with some changes in base class?

Reply:
I've redone StaticShapeNMS to be inherited from NonMaxSuppressionIE3 which is the same as NMS-5 but propagating the upper-bound for its outputs so now this code is not needed. Removed.